### PR TITLE
zkp changes to support extended field element use by zirgen codegen

### PR DESCRIPTION
**Pull request by shkoo**
* add Shift as a poly ext step opcode to return {0, 1, 0...}
* Implement a more complete set of operators for field elements
* Add debugging capabilities to PolyExtStep

### CODE REVIEW

File: risc0/zkp/src/adapter.rs [hunk 1]
```diff
@@ -210,45 +212,74 @@ impl PolyExtStep {
...
             PolyExtStep::Add(x1, x2) => {
-                fp_vars.push(fp_vars[*x1] + fp_vars[*x2]);
+                let val = fp_vars[*x1] + fp_vars[*x2];
+                tracing::trace!("[{}] {self:?} -> {val:?}", fp_vars.len());
+                fp_vars.push(val);
             }
             PolyExtStep::Sub(x1, x2) => {
-                fp_vars.push(fp_vars[*x1] - fp_vars[*x2]);
+                let val = fp_vars[*x1] - fp_vars[*x2];
+                tracing::trace!("[{}] {self:?} -> {val:?}", fp_vars.len());
```
Comment (flaub) [inline risc0/zkp/src/adapter.rs:235]: If these aren't needed anymore, let's comment them out.

**Comment by flaub** [comment]
Do we need corresponding changes to cuda/metal?

File: risc0/zkp/src/adapter.rs (hunk 1 above)
Comment (shkoo) [reply risc0/zkp/src/adapter.rs:235]: I put them behind a const so they're easy to re-enable.  Does that work?

**Comment by shkoo** [comment]
> Do we need corresponding changes to cuda/metal?

I don't think so; best I can tell, poly ext only gets run on CPU

**Review by flaub — APPROVED** [review]

**Pull request merged**